packages/chain/node_test.go

```
@@ -43,34 +43,37 @@ type tc struct {
 	n        int
 	f        int
 	reliable bool
+	timeout  time.Duration
 }
 
-func TestBasic(t *testing.T) {
+func TestNodeBasic(t *testing.T) {
 	t.Parallel()
 	tests := []tc{
-		{n: 1, f: 0, reliable: true},  // Low N
-		{n: 2, f: 0, reliable: true},  // Low N
-		{n: 3, f: 0, reliable: true},  // Low N
-		{n: 4, f: 1, reliable: true},  // Minimal robust config.
-		{n: 10, f: 3, reliable: true}, // Typical config.
+		{n: 1, f: 0, reliable: true, timeout: 10 * time.Second},   // Low N
+		{n: 2, f: 0, reliable: true, timeout: 20 * time.Second},   // Low N
+		{n: 3, f: 0, reliable: true, timeout: 30 * time.Second},   // Low N
+		{n: 4, f: 0, reliable: true, timeout: 40 * time.Second},   // Minimal robust config.
+		{n: 4, f: 1, reliable: true, timeout: 50 * time.Second},   // Minimal robust config.
+		{n: 10, f: 3, reliable: true, timeout: 130 * time.Second}, // Typical config.
 	}
 	if !testing.Short() {
 		tests = append(tests,
-			tc{n: 4, f: 1, reliable: false},  // Minimal robust config.
-			tc{n: 10, f: 3, reliable: false}, // Typical config.
-			tc{n: 31, f: 10, reliable: true}, // Large cluster, reliable - to make test faster.
+			// TODO these "unreliable" tests are crazy, they either succeed in 10~20s or run forever...
```

Don't comment them out. This can hide error.
Have you tried to increase retry periods?

---

These hang for a really really really long time. I saw the same issue happening in develop. Doesn't seem like a good test to have if we want to "speed up tests"

---

But these tests are essential. They check how the chain copes with connection drops, message reorderings, duplicated messages, etc.